aspnetcore/mvc/models/validation.md

```diff
@@ -57,6 +57,18 @@ MVC supports any attribute that derives from `ValidationAttribute` for validatio
 
 There may be instances where you need more features than built-in attributes provide. For those times, you can create custom validation attributes by deriving from `ValidationAttribute` or changing your model to implement `IValidatableObject`.
 
+## Notes on the use of the Required attribute
+
+Non-nullable [value types](/dotnet/csharp/language-reference/keywords/value-types) (such as `decimal`, `int`, `float`, and `DateTime`) are inherently required and don't need the `Required` attribute. The app performs no validation checks server-side for non-nullable types that are marked `Required`.
+
+MVC model binding, which isn't concerned with validation and validation attributes, rejects a form field submission containing only whitespace for a non-nullable type. Model binding usually ignores missing data for non-nullable types.
+
+Client-side validation always requires a value for a form field that corresponds to a C# property marked `Required`. `Required` can be used to control the client-side validation error message.
```

Comment:
Good points but please add a middle sentence such as "MVC also always requires a value for a form field that corresponds to a C# property of a non-nullable type." And, start or end the last sentence with "in both cases".

Reply:
> wordsmithed variant

Yes ... because the topic of the Line 64 paragraph is model binding; the topic of the next paragraph is client-side validation. I'd like to keep the reader's mind focused on model binding in the former, while engaging them on client-side behavior in the latter.

This sentence in the Line 64 paragraph ...

> MVC model binding ... rejects a form field submission containing only whitespace for a non-nullable type.

... expresses a similar concept to ...

> MVC also always requires a value for a form field that corresponds to a C# property of a non-nullable type.

I wonder if we can combine the points on whitespace and a missing form field value. The last sentence should probably also explain that "missing data" means an entirely missing representation of a form field in the incoming data.

> MVC model binding, which isn't concerned with validation and validation attributes, rejects a form field submission containing a missing value or whitespace for a non-nullable type. Model binding usually ignores missing data for non-nullable types, where the form field is absent from the incoming form data.